cloudcix_primitives/storage_secondary_lxd.py

```diff
@@ -0,0 +1,291 @@
+        volume_name:
+            description: The name of the LXD storage volume to attach.
+            type: string
+            required: true
+        mount_point:
+            description: The mount point for the volume inside the container.
+            type: string
+            required: true
+        verify_lxd_certs:
+            description: Boolean to verify LXD certs.
+            type: boolean
+            required: false
+        storage_pool:
+            description: The name of the storage pool containing the volume.
+            type: string
+            required: false
+            
+    return:
+        description: |
+            A tuple with a boolean flag indicating success or failure and a message.
+        type: tuple
+    """
+    # Define messages
+    messages = {
+        1000: f'Successfully attached volume {volume_name} to container {container_name} on {endpoint_url}',
+        1001: f'Volume {volume_name} is already attached to container {container_name}',
+        3021: f'Failed to connect to {endpoint_url} for containers.get payload',
+        3022: f'Failed to get container {container_name} configuration',
+        3023: f'Failed to attach volume {volume_name} to container {container_name}. Error: ',
+    }
+
+    def run_host(endpoint_url, prefix, successful_payloads):
+        rcc = LXDCommsWrapper(comms_lxd, endpoint_url, verify_lxd_certs, project)
+        fmt = HostErrorFormatter(
+            endpoint_url,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+
+        # Get the container
+        ret = rcc.run(cli='containers.get', name=container_name)
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, f"{prefix+1}: {messages[prefix+1]}"), fmt.successful_payloads
+        if ret["payload_code"] != API_SUCCESS:
+            return False, fmt.payload_error(ret, f"{prefix+2}: {messages[prefix+2]}"), fmt.successful_payloads
+
+        container = ret['payload_message']
+        fmt.add_successful('container get', ret)
+
+        # Check if the volume is already attached
+        if volume_name in container.devices:
+            # Volume already attached
+            return True, f'1001: {messages[1001]}', fmt.successful_payloads
+
+        # Create device config
+        device_config = {
+            "type": "disk",
+            "pool": storage_pool,
+            "source": volume_name,
+            "path": mount_point,
+        }
+
+        # Add the device to the container
+        container.devices[volume_name] = device_config
+
+        # Update the container configuration
+        try:
+            container.save(wait=True)
+            fmt.add_successful('container.save', {'device_added': volume_name})
+        except Exception as e:
+            return False, f"{prefix+3}: {messages[prefix+3]}{e}", fmt.successful_payloads
+        
+        return True, '', fmt.successful_payloads
```

Comment:
```suggestion
        return True, f'1000: {messages[1000]}', fmt.successful_payloads
```

Returning message code here so final return statement can return `1000` (attached successfully) or `1001` (already attached)